### PR TITLE
Osquerybeat: Configuration change to allow support of osquery configuration features 1:1

### PR DESCRIPTION
**Pull request by aleksmaus**
## What does this PR do?

Support the new policy configuration format. This allows to pass osquery configuration pretty much as is with the policy, no longer mapping the queries to the input streams. 
This change also relies on the new osquery_manager integration plugin limiting one integration per policy, similar to what Endpoint does. This solves the problem of osquery startup configuration definition, since now it is at the integration level.
Going forward this change allows us to easily to surface any osquery feature right out of the box with little or no changes to osquerybeat.
This introduces a set of protected options internally that user should not be able to change.

According to the documentation some osquery configuration options might not require restart, but in practice it looks like some options like ```disable_tables``` for example if provided through the configuration plugin does not take effect unless it is provided as CLI flag or through the flagsfile. Thus for any osquery option change CLI or not, the osquery is restarted and the options are provided as CLI flags. 

For example, the simple integration inputs configuration that allows to enable process events collection:
```
    "inputs": [
        {
            "type": "osquery",
            "enabled": true,
            "streams": [],
            "policy_template": "osquery_manager",
            "config": {
                "osquery": {
                    "value": {
                        "options": {
                            "disable_events": false,
                            "disable_audit": false,
                            "audit_allow_config": true,
                            "audit_allow_process_events": true
                        }
                    }
                }
            }
        }
    ]
```

Here is another example of the integration configuration payload body, this defines the global scheduled query, the pack
and the option to disable the ```users```  table:
```
{
    "name": "osquery_manager-1",
    "description": "",
    "namespace": "default",
    "policy_id": "85cf0340-1492-11ec-b58e-633cbe5c70b4",
    "enabled": true,
    "output_id": "",
    "package": {
        "name": "osquery_manager",
        "title": "Osquery Manager",
        "version": "0.6.0"
    },
    "inputs": [
        {
            "type": "osquery",
            "enabled": true,
            "streams": [],
            "policy_template": "osquery_manager",
            "config": {
                "osquery": {
                    "value": {
                        "schedule": {
                            "macos_uptime": {
                                "query": "SELECT * FROM uptime",
                                "interval": 60
                            }
                        },
                        "options": {
                            "disable_tables": "users"
                        },
                        "packs": {
                            "internal_stuff": {
                                "queries": {
                                    "users": {
                                        "query": "SELECT * FROM users limit 5",
                                        "interval": 90
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    ]
}
```

The osquerybeat changes are backwards compatible and will work with the exiting 7.x inputs/stream configuration. Going forward Kibana code can be adjusted to still support reading the integration configuration from streams, but update to the new format upon save @patrykkopycinski 

## Why is it important?

Solves the problem of "global"/runtime osquery configuration definition. Provides better solution for adopting all of osquery features going forward.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas

## Related issues

The new configuration would be only allowed with the following integration package change:
- Relates https://github.com/elastic/integrations/pull/1715

## Screenshots

Process events:
Screen Shot 2021-09-13 at 9 47 49 AM

File events:
Screen Shot 2021-09-13 at 9 29 54 AM

### CODE REVIEW

File: x-pack/osquerybeat/beater/config_plugin.go [hunk 1]
```diff
@@ -18,9 +17,9 @@ import (
 )
 
 const (
-	configName           = "osq_config"
-	scheduleSplayPercent = 10
-	maxECSMappingDepth   = 25 // Max ECS dot delimited key path, that is sufficient for the current ECS mapping
+	configName                  = "osq_config"
+	defaultScheduleSplayPercent = 10
```
Comment (james-elastic) [inline x-pack/osquerybeat/beater/config_plugin.go:21]: What does this do?

File: x-pack/osquerybeat/beater/config_plugin.go [hunk 2]
```diff
@@ -169,57 +151,127 @@ func (p *ConfigPlugin) render() (string, error) {
...
+		p.queriesCount = queriesCount
+	}()
+
+	// Return if no inputs, all the members will be reset by deferred call above
+	if len(inputs) == 0 {
+		return nil
+	}
+
+	// Since 7.16 version only one integration/input is expected
+	// The inputs[0].Osquery can be nil if this is pre 7.16 integration configuration
```
Comment (james-elastic) [inline x-pack/osquerybeat/beater/config_plugin.go:180]: good comment :)

**Review by james-elastic — APPROVED** [review]

File: x-pack/osquerybeat/beater/config_plugin.go (hunk 1 above)
Comment (aleksmaus) [reply x-pack/osquerybeat/beater/config_plugin.go:21]: Value in percent from 1-100 to splay the queries execution interval:

https://github.com/osquery/osquery/blob/2cd5b42c8f8fa52c6d251c6537595c7c59c90f4c/tools/deployment/osquery.example.conf#L22

https://github.com/osquery/osquery/blob/d2be385d71f401c85872f00d479df8f499164c5a/osquery/config/packs.h#L162

It's set to percent 1-100 .... lots of places in oquery source code and kolide launcher set it to 10by default

Comment (james-elastic) [reply x-pack/osquerybeat/beater/config_plugin.go:21]: Thanks!

**Review by scunningham — APPROVED** [review]
look sgood

**Pull request merged**